core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -156,13 +160,42 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit {
 
     // Update on the server.
     this.workflowExecutionsService
-      .setIsBookmarked(this.workflow.wid, row.eId, !wasPreviouslyBookmarked)
+      .groupSetIsBookmarked(this.workflow.wid, [row.eId], wasPreviouslyBookmarked)
       .pipe(untilDestroyed(this))
       .subscribe({
         error: (_: unknown) => (row.bookmarked = wasPreviouslyBookmarked),
       });
   }
 
+  allBookmarkCheck(rows: Set<WorkflowExecutionsEntry>): boolean {
+    if (rows !== undefined) {
+      for (let row of rows) {
```

Comment:
Please do the check avoiding this loop for example use contains() or any other function to quickly check if any row satisfies your condition